packages/rest-client-jsonapi/src/services/utils/resources/resources.test.ts

```diff
@@ -159,3 +164,57 @@ describe("rest-client-jsonapi/services/utils/resources", () => {
     })
   })
 })
+
+describe("hatchifyRelationshipsToJsonApiRelationship", () => {
+  it("Correctly converts relationship objects with one or many relationships", () => {
+    const schema: Schema = {
+      name: "Article",
+      displayAttribute: "name",
+      attributes: {
+        title: "string",
+        body: "string",
+      },
+      relationships: {
+        person: {
+          type: "one",
+          schema: "Person",
+        },
+        tag: {
+          type: "many",
+          schema: "Tag",
+        },
+      },
+    }
+    const sourceConfig = {
+      baseUrl: "http://localhost:3000/api",
+      schemaMap: {
+        Article: { type: "article", endpoint: "articles" },
+        Person: { type: "person", endpoint: "people" },
+        Tag: { type: "tag", endpoint: "tags" },
+      },
+    }
+
+    const relationships: Record<
+      string,
+      | Omit<ResourceRelationship, "__schema">
+      | Array<Omit<ResourceRelationship, "__schema">>
+    > = {
+      person: { id: "1" },
+      tag: [{ id: "1" }, { id: "2" }],
+    }
+
+    const expected: Record<string, JsonApiResourceRelationship> = {
+      person: { data: { id: "1", type: "Person" } },
+      tag: {
+        data: [
+          { id: "1", type: "Tag" },
+          { id: "2", type: "Tag" },
+        ],
```

Comment:
the `type` in the expected should map to `sourceConfig.schemaMap[schema].type`, so in this case we expected them to be `tag` and `person`.

just as a note: the schemaMap config allows people to override types (in case of jsonapi type being different than schema name), if they don't pass it in (when using jsonapi) it defaults `type` to `schemaName`